### ISSUE TITLE
netlist activate: "HTTP 405 Method Not Allowed"

### ISSUE THREAD

**Issue by jfrome** [apiheat/akamai-cli-netlist#9]
akamai netlist --version = 5.0.2

When I attempt to activate a network list I receive at "HTTP 405 Method Not Allowed".  Debug output in this gist: [HTTP 405 Method Not Allowed](https://gist.github.com/jfrome/0b0473d89cbf278c8e2e805ddfdd895e)

I created and added subnets to this netlist with the same API credentials/config, so I don't believe there is a permissions/config issue.

**Comment by RafPe** [comment]
Hey,

We have seen similar output once when working with akamai apis. In our case there were some problems on Akamai side. 

If u would try to activate it again with cli and see if u continue to get errors. 

If so - we can then try to reach out to Akamai to ask some questions :)

**Comment by jfrome** [comment]
This has been an ongoing issue for a few weeks, just with activating network lists.  I have opened a case with Akamai.  I'll re-open issue if there is anything additional needed.  Thanks.

**Comment by jfrome** [comment]
Akamai responded that it appears that the call to activate was initiated with a DELETE method instead of a POST:
![2019-03-27 07_57_58-DELETE_error](https://user-images.githubusercontent.com/540476/55082071-9cdb1300-5066-11e9-851a-cff8739a7f4a.png)

Any idea why this would be called using a DELETE?

**Comment by RafPe** [comment]
@jfrome you are spot on the problem 👍  Please download newest version .. 

Root cause can be seen here https://github.com/apiheat/go-edgegrid/issues/24

Let us know if it fixed the problem ;)

**Comment by jfrome** [comment]
I have updated to latest netlist (5.0.3), but am still seeing the error.  Even on a fresh install, it looks like the latest apiheat/go-edgegrid is not getting pulled in:

```bash
> pwd
/home/josh/.akamai-cli/pkg/dep/sources/https---github.com-apiheat-go--edgegrid

> grep -n  MethodDelete *.go                                                                                                                                                                                                              
v2-network_list.go:272: resp, err := nls.client.makeAPIRequest(http.MethodDelete, path, qParams, &respStruct, nil, nil)
v2-network_list.go:298: resp, err := nls.client.makeAPIRequest(http.MethodDelete, path, qParams, &respStruct, nil, nil)
v2-network_list.go:349: resp, err := nls.client.makeAPIRequest(http.MethodDelete, path, qParams, &respStruct, nil, nil)
```

**Comment by RafPe** [comment]
:) `lock` file did not get refreshed during deploy. Travis will build `v5.0.4` and if you could download that one it has the correct edge client 

```
> [SHELL]  RafPe $ grep -n  MethodDelete github.com/apiheat/go-edgegrid/v2-network_list.go
272:	resp, err := nls.client.makeAPIRequest(http.MethodDelete, path, qParams, &respStruct, nil, nil)
349:	resp, err := nls.client.makeAPIRequest(http.MethodDelete, path, qParams, &respStruct, nil, nil)
```

**Comment by jfrome** [comment]
Getting closer :)  Using current version (5.0.5), now getting HTTP400 'Bad Request' error.  See debug output here: https://gist.github.com/jfrome/bf3ab63edd9687c5ff4589c42754540b

**Comment by RafPe** [comment]
@jfrome thats interesting. Could you highlight what actions are you taking and in which order ? i.e. creating new network list / adding items / activating etc ?

**Comment by RafPe** [comment]
Ok @jfrome you can safely ignore my previous comment. I found the problem .... For unknown reason edge client was not adding request body. 

In debug I should have noticed 

```
DEBU[0000] [NewRequest]::Object request provided type ( %!s(<nil>) )
DEBU[0000] [NewRequest]::Body object added to request
```

It should look like 

```
DEBU[0000] [NewRequest]::Method is POST
DEBU[0000] [NewRequest]::Marshal request object
DEBU[0000] [NewRequest]::Object request provided type ( edgegrid.NetworkListActivationOptsv2 )
DEBU[0000] [NewRequest]::Body object added to request
DEBU[0000] [NewRequest]::Body object is:{"comments":"","notificationRecipients":["me@rafpe.ninja"],"fast":false}
```

I released `v5.0.6` which I tested with my API credentials and it worked like a charm :)  Try it and let us know if it works 4u now

**Comment by jfrome** [comment]
Good news! 5.0.6 works.  I had to update Gopkg.toml to get latest version of go-edgegrid and compile manually, but I am now able to activate network lists.  Thanks!
